# RLSDSPCA

## (I) Datasets
multisource gene expression dataset is a four category dataset and is composed of three different cancer types of gene expression datasets, including cholangiocarcinoma (CHOL), head and neck squamous cell carcinoma (HNSCC), and pancreatic adenocarcinoma (PAAD) from TCGA. The CHOL dataset included 45 samples (9 normal samples, 36 cancer samples) and 20,502 genes; the HNSCC dataset consisted of 418 samples (20 normal samples, 398 cancer samples) and 20,502 genes; and the PAAD dataset contained 180 samples (4 normal samples, 176 cancer samples) and 20,502 genes. While COAD dataset is a two category dataset with 20502 genes, including 262 colon adenocarcinoma data samples and 19 normal tissue samples.
In this study, we used this multisource gene expression dataset and the COAD dataset as the benchmark datasets to perform the experiments of characteristic gene selection and tumor classification.
The datasets of gene expression for characteristic gene selection and tumor classification were obtained from The Cancer Genome Atlas (TCGA, https://portal.gdc.cancer.gov/) database.
## (II) Model file
PCA,gLPCA,gLSPCA,RgLPCA,SDSPCA. The code of the comparision methods and featureselection can be obtained in model file
## (III)  plot file
The code of figure plot can be obtained in plot file
## (IV) methods references
#### [1] I. Jolliffe, “Principal component analysis,” Technometrics, vol. 45, no. 3, pp. 276, 2003.
#### [2]C.-M. Feng, Y. Xu, J.-X. Liu, Y.-L. Gao, and C.-H. Zheng, “Supervised discriminative sparse PCA for com-characteristic gene selection and tumor classification on multiview biological data,” IEEE transactions on neural networks and learning systems, vol. 30, no. 10, pp. 2926-2937, 2019.
#### [3]B. Jiang, C. Ding, and J. Tang, "Graph-Laplacian PCA: Closed-form solution and robustness." pp. 3492-3498.
#### [4]C.-M. Feng, Y. Xu, M.-X. Hou, L.-Y. Dai, and J.-L. Shang, “PCA via joint graph Laplacian and sparse constraint: Identification of differentially expressed genes and sample clustering on gene expression data,” BMC bioinformatics, vol. 20, no. 22, pp. 1-11, 2019.
## (V) Contact 
If you have any suggestions/questions about the work, PLEASE contact with us. E-mail: zhangluxing@njust.edu.cn
